### ISSUE TITLE
Remove accessory is not working properly

### ISSUE THREAD

**Issue by mkudrnac** [ikalchev/HAP-python#4]
I found a small issue with removing accessory.
When I try to remove it  I takes very long time sometimes iPhone get stuck and in console there is this error and accessory.pickle is not removed

screenshots:
![screen shot 2017-11-29 at 19 39 16](https://user-images.githubusercontent.com/12230741/33392794-5f58e28c-d53d-11e7-8d6f-5ca5cd230249.png)

before remove:
![img_0937](https://user-images.githubusercontent.com/12230741/33392827-7b6b4d5c-d53d-11e7-841d-33b9e8208f17.png)

after remove:
![img_0938](https://user-images.githubusercontent.com/12230741/33392838-831c132e-d53d-11e7-899f-64bd48670214.png)

I found that there is ```# TODO: can we have more than one connection?``` in my home I have AppleTV as HUB and another iPhone with I'm sharing my home.

Dont worry to ask me with some help :-)

**Comment by ikalchev** [comment]
Yes, I am aware of this, although I don't get a KeyError on unpair (just a very long waiting time, as in your case). I just figured that this same issue leads to lost updates/replies - they get delivered to the last client that has connected. This in turn makes the iOS with the lost update unhappy and marks the accessory as "not responding".

I'm quite confident that you are right - this TODO should be the culprit. I expect it won't take long to fix.

About the pickle: I think the problem is that the new state is not pickled and not that the pickle is not deleted. Nevertheless, it's an issue and should be solved by either of the above. Preferences?

So to sum up the issues:
 - Having only a single client can lead to crashes and being unable to handle unpair requests.
 - Replies and updates are delivered to the last connected client.
 - accessory.pickle is not updated/deleted on unpair.

**Comment by mkudrnac** [comment]
Yes you are right, this is not so critical bug, only just not user friendly and for my testing I have to shutdown server and then remove accessory and delete pickle, takes a lot of time :-)

In KhaosT/HAP-NodeJS unpair is working even with more connected devices maybe you can look for some inspiration :-)

**Comment by ikalchev** [comment]
You mean that it unpairs a single device, right?

It's quite the bug - lost updates are not nice. The unpairing thing is not that bad, I agree.

Yeah, it was the source of inspiration many times. Thanks :) I still think it should be fairly easy to fix this.

**Comment by mkudrnac** [comment]
When you remove accessory It should unpair it from all devices, I don't think that there is option to unpair only one device.

**Comment by ikalchev** [comment]
Could you please get latest and see if the bug appears again? I tested with two clients, both of which received updates simultaneously. Also, unpairing does not hang the app and you can then pair again.

Note that when you remove an accessory from one device, other devices will get unpaired, but will still receive updates from the accessory until the removal of the accessory gets synched among them. This is as HAP intended.

I had a closer look at the first screenshot you posted. You hit an interesting race condition: While one device was negotiating a session, another device removed the accessory from the Home and issued requests to unpair all devices that share the Home. So when the accessory needed the public key of the first device to negotiate the session, this turned out to be removed during the unpairing (hence the KeyError). This is now fixed.

Anyway, I hope this issue is resolved. Let me know :)

**Comment by mkudrnac** [comment]
Well I must thank you again now it is working properly and iPhone do not get stuck

According to IP 192.168.1.10 that device is AppleTV which is a HUB and provides remote access and automations. (Synced through iCloud I guess)

Here is console screenshot and everything is woking ok
![screen shot 2017-11-29 at 23 18 37](https://user-images.githubusercontent.com/12230741/33402066-b32ffa9e-d55b-11e7-9e0e-3326fb6a1da5.png)

**Comment by ikalchev** [comment]
Awesome, Miroslav, nothing to thank for. Keep raising whatever issues you find and any suggestions you may have :)

PS I will get back to you for the sensor repos soon.
